docs/articles/clustering/cluster-sharding.md

```diff
@@ -257,3 +257,188 @@ In the normal operation of an Akka.NET cluster, the sharding system automaticall
+    
+        // Register a callback for the "cluster is up" event
+        var cluster = Cluster.Get(system);
+        cluster.RegisterOnMemberUp(() =>
+        {
+            ProduceMessages(system, shardRegionProxy);
+        });
+        break;
+    
+    case BackEndRole:
+        await sharding.StartAsync(
+            typeName: "customer",
+            entityPropsFactory: e => Props.Create(() => new Customer(e)),
+            settings: ClusterShardingSettings.Create(system).WithRole(BackEndRole), 
+            messageExtractor: new MessageExtractor(10));
+        break;
+}
+```
+
+### Sending Messages To the Sharded Actors
+
+Finally we can start sending messages to the sharded actors through the proxy coordinator actor.
+
+```c#
+private static void ProduceMessages(ActorSystem system, IActorRef shardRegionProxy)
+{
+    var customers = new[] { "Yoda", "Obi-Wan", "Darth Vader", "Princess Leia"};
+    var items = new[] { "Yoghurt", "Fruits", "Lightsaber", "Fluffy toy", "Dreamcatcher"};
+
+    system.Scheduler.Advanced.ScheduleRepeatedly(TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(3), () =>
+    {
+        var customer = PickRandom(customers);
+        var item = PickRandom(items);
+        
+        var message = new ShardEnvelope(customer, new Customer.PurchaseItem(item));
+        shardRegionProxy.Tell(message);
+    });
+}
+```
+
+Note that the message need to be wrapped inside an envelope so that cluster sharding will know where
+to send said message to the correct shard and actor.
+
+### Sharded Actor
+
+Sharded actors are usually persisted using `Akka.Persistence` so that it can be restored after it is
+terminated, but a regular `ReceiveActor` would also work. In this example, we will be using a regular
+`ReceiveActor` for brevity. For a distributed data backed cluster sharding example, please see
+[this example](https://github.com/akkadotnet/akka.net/tree/dev/src/examples/Cluster/ClusterSharding/ClusterSharding.Node)
+in the GitHub repository.
+
+```c#
+public class Customer : ReceiveActor
+{
+    public sealed class PurchaseItem
+    {
+        public readonly string ItemName;
+
+        public PurchaseItem(string itemName)
+        {
+            ItemName = itemName;
+        }
+    }
+    
+    private readonly List<string> _purchasedItems = new List<string>();
+
+    public Customer(string persistenceId)
+    {
+        Receive<PurchaseItem>(purchase =>
+        {
+            _purchasedItems.Add(purchase.ItemName);
+            var name = Uri.UnescapeDataString(Self.Path.Name);
+            Console.WriteLine(
+                $"'{name}' purchased '{purchase.ItemName}'.\nAll items: [{string.Join(", ", _purchasedItems)}]\n--------------------------");
+        });
+    }
+}
+```
+
+### Message Envelope and Message Extractor
+
+The shard coordinator would need to know which shard and entity it needs to send the messages to,
+we do that by embedding the entity and shard id information in the envelope we send the message in.
```

Comment:
> shard id information in the envelope we send the message in

shard id information in the messages we send to the it. In this case, we're using a custom `ShardEnvelope` message type to wrap all of our messages.